docs/changelog/2107.md

```diff
@@ -0,0 +1 @@
+- Remove columns "QNColumns", "DeletedQNColumns", and "DroppedQNColumns" from `*-iterations.log` for non-quasi-Newton acceleration schemes.
```

Comment:
The current state doesn't remove the columns

Reply:
I guess because of this part?
```
if (_acceleration) {
  _iterationsWriter->addData("QNColumns", io::TXTTableWriter::INT);
  _iterationsWriter->addData("DeletedQNColumns", io::TXTTableWriter::INT);
  _iterationsWriter->addData("DroppedQNColumns", io::TXTTableWriter::INT);
}
```
Anyway: I will revert this PR to the non-breaking version and put the suggestions that would be breaking into an issue.